test/PSRP.Tests.ps1

```diff
@@ -0,0 +1,138 @@
+Describe " PowerShell Remoting basic functional tests" -Tag @("CI") {
+        BeforeALL {
+                    $LinuxHostName = $env:LINUXHOSTNAME
+                    $WindowsHostName = $env:WINDOWSHOSTNAME
+                    $passwordString = $env:PASSWORDSTRING
+                    $badUserName = "badUserName"
+                    $badPassword = "badPassword"
```

Comment:
There is a lot of duplicate code. It could be refactored to not care as much about what platform it is running on since the same code is run and only the parameters change. That said Mac and Linux do not have winrm so omicli has to be used.